### PR TITLE
Option to launch Dynamo directly without splash screen.

### PR DESCRIPTION
**Pull request by reddyashish**
### Purpose

Option to launch Dynamo directly without splash screen. This is a follow-up of https://github.com/DynamoDS/DynamoRevit/pull/2913 after enabling splash screen in Revit.
@BogdanZavu reported that player samples were not being edited with the latest changes.

When Dynamo player samples are edited, main dynamo is needed and it should start a example workspace. So we need a option to launch main Dynamo directly in that case.

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.

### Reviewers
@QilongTang @Mikhinja

### CODE REVIEW

**Comment by QilongTang** [comment]
@mjkkirschner There is already a setting in Dynamo for this. FYI: @reddyashish

**Comment by reddyashish** [comment]
That setting is for static splash screen. Dynamic splash screen is always shown. 
@mjkkirschner you mean an option to load the Dynamic main window and not the dynamic splash screen from Core? 
Yes that would be ideal. I thought, since splash screen is a webview2 component and not inside core, DynamoRevit was linked to splash screen directly.

**Comment by QilongTang** [comment]
> hey @reddyashish @QilongTang -nice that this is possible - but do you think it would be better if this was just an option in DynamoCore instead so that integrators would not have to do anything special to accomplish this? (maybe an improvement for the future)

I feel like I want your elaboration on this

**Comment by mjkkirschner** [comment]
yeah, I don't have a solution in mind, it just feels to me that the splash screen is too complex, and using it has become quite complex for integrators, and we should redesign the API for using it so all they need to think about is - I want to start dynamo with a splash screen or without.

**Comment by QilongTang** [comment]
@mjkkirschner That make sense. @reddyashish It looks like to me, that a setting is needed to totally skip even the static splash. But it might be too late for 2.18 release. Also can you attach a gif about the behavior before and after?

**Comment by mjkkirschner** [comment]
yeah agree regarding 2.18 - this is more a comment in general about splash screen / API consideration in general for the future.

**Comment by reddyashish** [comment]
Before a lot of splash screens were starting and it would crash. 

splash D4R

File: src/DynamoRevit/DynamoRevit.cs
Comment (QilongTang) [inline src/DynamoRevit/DynamoRevit.cs:308]: `when ShowUiKey is false`?

File: src/DynamoRevit/DynamoRevit.cs
Comment (QilongTang) [inline src/DynamoRevit/DynamoRevit.cs:316]: It seems the catch part is common between both condition, should we put the whole logic into try-catch so we only have one single try catch?

File: src/DynamoRevit/DynamoRevit.cs
Comment (QilongTang) [inline src/DynamoRevit/DynamoRevit.cs:482]: I guess this function simply reflects logic before we enable the splash screen in D4R right?

Comment (reddyashish) [reply src/DynamoRevit/DynamoRevit.cs:482]: yes

File: src/DynamoRevit/DynamoRevit.cs
Comment (reddyashish) [reply src/DynamoRevit/DynamoRevit.cs:308]: when ShowUiKey is false, it will go to else block in line 392(new file). We are changing the default parameter in CheckJournalForKey() to check if that flag is true or false or not present.

File: src/DynamoRevit/DynamoRevit.cs
Comment (reddyashish) [reply src/DynamoRevit/DynamoRevit.cs]: Yes changed it, now we have a single try-catch block.

**Review by Mikhinja — APPROVED** [review]

**Comment by reddyashish** [comment]
@Mikhinja Can you merge this if it looks good to you. I don't have write access.

**Pull request merged**
